snap/component.go

```diff
@@ -41,16 +54,85 @@ type ComponentSideInfo struct {
 	Revision  Revision            `json:"revision"`
 }
 
+// NewComponentSideInfo creates a new ComponentSideInfo.
+func NewComponentSideInfo(cref naming.ComponentRef, rev Revision) *ComponentSideInfo {
+	return &ComponentSideInfo{
+		Component: cref,
+		Revision:  rev,
+	}
+}
+
+// componentPlaceInfo holds information about where to put a component in the
+// system. It implements ContainerPlaceInfo and should be used only via this
+// interface.
+type componentPlaceInfo struct {
+	// Name and revision for the component
+	compName     string
+	compRevision Revision
+	// snapInstance and SnapRevision identify the snap that uses this component.
+	snapInstance string
+	snapRevision Revision
+}
+
+var _ ContainerPlaceInfo = (*componentPlaceInfo)(nil)
+
+// MinimalComponentContainerPlaceInfo returns a ContainerPlaceInfo with just
+// the location information for a component of the given name and revision that
+// is used by a snapInstance with revision snapRev.
+func MinimalComponentContainerPlaceInfo(compName string, compRev Revision, snapInstance string, snapRev Revision) ContainerPlaceInfo {
```

Comment:
nitpick: I still have a slight preference for this to take a PlaceInfo instead of snapInstance/snapRev